code/modules/clothing/clothing.dm

```diff
@@ -137,6 +147,10 @@
 			if(variable in user.vars)
 				user_vars_remembered[variable] = user.vars[variable]
 				user.vars[variable] = user_vars_to_edit[variable]
+	if(greyscale)
+		if(ishuman(user))
+			var/mob/living/carbon/human/H = user
+			H.regenerate_icons()

```

Comment:
More specifically, I don't believe this is even necessary, as just putting on the item will already call the necessary procs to add the greyscale overlay.